src/base/src/db3_document.rs

```diff
@@ -0,0 +1,197 @@
+use bson::spec::BinarySubtype;
+use bson::Document;
+use bson::{Binary, Bson, RawDocumentBuf};
+use db3_error::{DB3Error, Result};
+use fastcrypto::traits::ToFromBytes;
+use serde_json::{Map, Value};
+
+#[derive(Debug)]
+pub struct DB3Document {
+    doc: Document,
+}
+impl DB3Document {
+    pub fn into_bytes(&self) -> Vec<u8> {
+        let row_doc = RawDocumentBuf::from_document(&self.doc).unwrap();
+        row_doc.into_bytes()
+    }
+    pub fn add_documentId(&mut self, docId: &str) {
+        self.doc.insert("_docId", docId);
+    }
+
+    pub fn get_documentId(&self) -> Option<&str> {
+        match self.doc.get_str("_docId") {
+            Ok(docId) => Some(docId),
+            Err(err) => None,
+        }
+    }
+    pub fn add_txId(&mut self, txId: &[u8]) {
+        self.doc.insert(
+            "_txId",
+            Bson::Binary(Binary {
+                subtype: BinarySubtype::Generic,
+                bytes: txId.to_vec(),
+            }),
+        );
+    }
+    pub fn get_txId(&mut self) -> Option<&Vec<u8>> {
+        match self.doc.get_binary_generic("_txId") {
+            Ok(txId) => Some(txId),
+            Err(err) => None,
+        }
+    }
```

Comment:
Whether will we  update the transaction id when a user updates the document

Reply:
Yep. Will will. It will be supported when we implement updateDocument operation